### PR TITLE
change bookinfo test to use helm install 

### PR DESCRIPTION
**Pull request by linsun**
since we have support for 2 ways to install istio (either generate the template then kubectl install or helm install), but the helm install path isn't tested in any automation test.

PR not ready to be reviewed -- open it so that I can test it (since no idea how to test it on my laptop)

### CODE REVIEW

File: tests/e2e/framework/kubernetes.go [hunk 1]
```diff
@@ -595,6 +622,72 @@ func (k *KubeInfo) deployIstio() error {
...
+		return err
+	}
+
+	// deploy tiller, helm cli is already available
+	if err := util.HelmInit("tiller"); err != nil {
+		log.Errorf("Failed to init helm tiller ")
+		return err
+	}
+	// wait till tiller reaches running
+	return util.CheckPodRunning("kube-system", "name=tiller", k.KubeConfig)
```
Comment (sebastienvas) [inline tests/e2e/framework/kubernetes.go]: did you mean to write "tiller" only ?

File: tests/e2e/framework/kubernetes.go [hunk 2]
```diff
@@ -333,63 +353,70 @@ func (k *KubeInfo) Teardown() error {
...
 		return nil
 	}
+	if *installer == "helm" {
+		// clean up using helm
+		err := util.HelmDelete("istio")
+		if err != nil {
+			return nil
+		}
+	} else {
+		if *useAutomaticInjection {
```
Comment (sebastienvas) [inline tests/e2e/framework/kubernetes.go]: I think that automatic injection is unrelated to helm install or kubectl install . @ayj  what do you think ?

Comment (linsun) [reply tests/e2e/framework/kubernetes.go]: I think we suppose automatic injection for helm, @ayj or @yusuoh ?

Comment (ayj) [reply tests/e2e/framework/kubernetes.go]: helm supports automatic injection (see https://github.com/istio/istio/blob/ca3049444afd3fc3170db9d7c7e47533be528eb7/install/kubernetes/helm/istio/values.yaml#L186).
